File: easy/2215. Find the Difference of Two Arrays/2215. Find the Difference of Two Arrays.py
from typing import List


class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        items1 = set(nums1)
        items2 = set(nums2)

        return [
            list(filter(lambda x: x not in items2, items1)),
            list(filter(lambda x: x not in items1, items2)),
        ]

File: easy/2215. Find the Difference of Two Arrays/2215. Find the Difference of Two Arrays.go
package main

func findDifference(nums1 []int, nums2 []int) [][]int {
	items1 := make(map[int]bool)
	items2 := make(map[int]bool)

	for _, num := range nums1 {
		items1[num] = true
	}

	for _, num := range nums2 {
		items2[num] = true
	}

	ans := make([][]int, 2)

	for i := range items2 {
		if !items1[i] {
			ans[1] = append(ans[1], i)
		}
	}

	for i := range items1 {
		if !items2[i] {
			ans[0] = append(ans[0], i)
		}
	}

	return ans
}
